src/cli/aspencli.py

```diff
@@ -349,6 +355,41 @@ def delete_samples(ctx, sample_ids):
     print(resp.text)
 
 
+@samples.command(name="update")
+@click.argument("sample_id")
+@click.option("--private-id", required=False, type=str, help="Update the sample private id")
+@click.option("--public-id", required=False, type=str, help="Update the sample public id")
+@click.option("--collection-date", required=False, type=str, help="Update the sample collection date")
+@click.option("--sequencing-date", required=False, type=str, help="Update the sample sequencing date")
+@click.option("--private", required=False, type=bool, help="Set whether the sample is private")
+@click.option("--location", required=False, type=int, help="Set the sample's collection location")
+@click.pass_context
+def update_samples(ctx, sample_id, private_id, public_id, collection_date, sequencing_date, private, location):
+    api_client = ctx.obj["api_client"]
+    if collection_date:
+        collection_date = dateparser.parse(collection_date).strftime('%Y-%m-%d')
```

Comment:
Hadn't heard of `dateparser` before, that's a neat library. I'm not sure it's necessary to bring it in as a dep though since we require our users to strictly adhere to YYYY-MM-DD format -- making ourselves follow the same convention when we use the CLI tool seems reasonable. Seems like the tradeoff here is adding an additional dependency VS more forgiving ingestion of dates when we use the tool. But since the request will just get turned down if we screw up the date formatting, it doesn't seem that beneficial to have the extra flexibility for date formats.

That said, it's just a local-only dependency for running our CLI tool, so it's definitely not a big deal.

Reply:
Yeah, good callout -- I agree that YYYY-MM-DD is pretty standard but I'm curious to see if having the flexibility to say "3 days ago" or "January 14 2021" might come in handy. Let's try it out for awhile and we can drop it if it we never need it.